internal/server/grpc_server_test.go

```diff
@@ -0,0 +1,99 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"testing"
+
+	"github.com/opendatahub-io/model-registry/internal/ml_metadata/proto"
+	"github.com/opendatahub-io/model-registry/internal/model/db"
+	"gorm.io/driver/sqlite"
+	"gorm.io/gorm"
+)
+
+func migrateDatabase(dbConn *gorm.DB) error {
+	// using only needed RDBMS type for the scope under test
+	err := dbConn.AutoMigrate(
+		db.Type{},
+	)
+	if err != nil {
+		return fmt.Errorf("db migration failed: %w", err)
+	}
+	return nil
+}
+
+func setup(tmpFile *os.File) (*gorm.DB, error) {
+	db, err := gorm.Open(sqlite.Open(tmpFile.Name()), &gorm.Config{})
+	if err != nil {
+		return nil, err
+	}
+	err = migrateDatabase(db)
+	if err != nil {
+		return nil, err
+	}
+	return db, nil
+}
+
+// Bare minimal test of PutArtifactType with a given Name
+func TestPutArtifactType(t *testing.T) {
+	f, err := os.CreateTemp("", "model-registry-db")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer os.Remove(f.Name())
+	db, err := setup(f)
+	if err != nil {
+		t.Errorf("Should expect DB connection: %v", err)
```

Comment:
`Should expect` in error messages sounds confusing. How about:

```suggestion
		t.Errorf("Expected DB connection: %v", err)
```